tests/ttnn/unit_tests/operations/test_all_reduce_t3000_frequent.py

```diff
@@ -0,0 +1,230 @@
+        per_chip_output_shape, math_op, mem_config, num_devices, num_links, input_dtype, layout
+    )
+    if is_known_failure:
+        pytest.skip(f"Skipping unsupported case {message}.")
+
+    mesh_device.enable_async(enable_async)
+    if enable_async:
+        logger.info(f"Using Async Mode for All Reduce Op Dispatch")
+
+    logger.info(f"Per chip output shape: {per_chip_output_shape}, devices: {num_devices}")
+    # Generate input tensors
+
+    tt_input_tensors = []
+    input_tensors = []
+    for i in range(num_devices):
+        input_tensor = torch.rand(per_chip_output_shape).bfloat16()
+        tt_input_tensors.append(
+            ttnn.Tensor(input_tensor, input_dtype)
+            .to(layout)
+            .to(mesh_device.get_device(mesh_device.get_device_ids()[i]), mem_config)
+        )
+        input_tensors.append(input_tensor)
+    unchunked_input_tensor = torch.cat(input_tensors)
+
+    assert len(tt_input_tensors) == num_devices
+
+    input_tensor_mesh = ttnn.aggregate_as_tensor(tt_input_tensors)
+    # Run the op
+    for i in range(num_iters):
+        output_tensor_mesh = ttnn.experimental.all_reduce(
+            input_tensor_mesh,
+            math_op=math_op,
+            num_links=num_links,
+            memory_config=mem_config,
+            topology=topology,
+        )
+
+        for device_id in mesh_device.get_device_ids():
+            ttnn.synchronize_device(mesh_device.get_device(device_id))
+        logger.info(f"Done iteration {i}")
+
+    tt_out_tensors = ttnn.get_device_tensors(output_tensor_mesh)
+    logger.info(f"Compare")
+    golden_canonical_out_tensor = torch.sum(unchunked_input_tensor, 0)
+
+    # Compare
+    mismatch = False
+    for i, t in enumerate(tt_out_tensors):
+        tt_output_tensor = t.cpu().to(ttnn.ROW_MAJOR_LAYOUT).to_torch()
+        eq, output = comp_pcc(tt_output_tensor, golden_canonical_out_tensor)
+        mismatch = mismatch or not eq
+        if not eq:
+            logger.error(f"output mismatch for tensor {i}")
+            if debug:
+                for w in range(tt_output_tensor.shape[0]):
+                    for z in range(tt_output_tensor.shape[1]):
+                        for y in range(tt_output_tensor.shape[2]):
+                            for x in range(tt_output_tensor.shape[3]):
+                                if tt_output_tensor[w, z, y, x] != golden_canonical_out_tensor[w, z, y, x]:
+                                    logger.error(
+                                        f"mismatch at {w}, {z}, {y}, {x}: {tt_output_tensor[w, z, y, x]} != {golden_canonical_out_tensor[w, z, y, x]}"
+                                    )
+
+        else:
+            logger.info(f"output match for tensor {i}")
+    assert not mismatch, f"{i} FAILED: {output}"
+
+
+@pytest.mark.timeout(120)
+@pytest.mark.parametrize(
+    "num_devices, num_links",
+    [
+        (8, 1),
+    ],
+)
+@pytest.mark.parametrize(
+    "per_chip_output_shape",
+    [
+        ([1, 1, 32, 4096]),
```

Comment:
Please add tests for all dimensions including the 0 and 1 too.
Please add tests for non-divisible by 32 dimensions or explicitly write that it is not supported.